HerPublicWebsite.BusinessLogic/ExternalServices/EmailSending/GovUkNotifyApi.cs

```diff
@@ -43,32 +45,41 @@ private EmailNotificationResponse SendEmail(GovUkNotifyEmailModel emailModel)
                 throw new EmailSenderException(EmailSenderExceptionType.Other);
             }
         }
-
-        public void SendReferenceNumberEmail(string emailAddress, string reference)
-        {
-            var template = govUkNotifyConfig.ApplicationReferenceNumberTemplate;
-            var personalisation = new Dictionary<string, dynamic>
+        
+        public void SendReferenceCodeEmail
+        (
+            string emailAddress,
+            string recipientName,
+            string referenceCode,
+            string custodianCode
+        ) {
+            var template = govUkNotifyConfig.ReferenceCodeTemplate;
+            LocalAuthorityData.LocalAuthorityDetails localAuthorityDetails;
+            try
             {
-                { template.ReferencePlaceholder, reference },
-                { template.MagicLinkPlaceholder, govUkNotifyConfig.BaseUrl + "returning-user/" + reference },
-                { template.ReturningUserLinkPlaceholder, govUkNotifyConfig.BaseUrl + "new-or-returning-user" },
-                { template.FeedbackLinkPlaceholder, Constants.FEEDBACK_URL_DEFAULT }
-            };
-            var emailModel = new GovUkNotifyEmailModel
+                localAuthorityDetails = LocalAuthorityData.LocalAuthorityDetailsByCustodianCode[custodianCode];
+            }
+            catch (KeyNotFoundException ex)
             {
-                EmailAddress = emailAddress,
-                TemplateId = template.Id,
-                Personalisation = personalisation
-            };
-            var response = SendEmail(emailModel);
-        }
-
-        public void SendRequestedDocumentEmail(string emailAddress, byte[] documentContents)
-        {
-            var template = govUkNotifyConfig.RequestDocumentTemplate;
+                logger.LogError
+                (
+                    ex,
+                    "Attempted to send reference code email with invalid custodian code \"{CustodianCode}\"",
+                    custodianCode
+                );
+                throw new ArgumentOutOfRangeException
+                (
+                    $"Attempted to send reference code email with invalid custodian code \"{custodianCode}\"",
+                    ex
+                );
+            }
+            
             var personalisation = new Dictionary<string, dynamic>
             {
-                { template.DocumentContentsPlaceholder, NotificationClient.PrepareUpload(documentContents) }
+                { template.RecipientNamePlaceholder, recipientName },
+                { template.ReferenceCodePlaceholder, referenceCode },
+                { template.LocalAuthorityNamePlaceholder, localAuthorityDetails.Name + " Council" },
```

Comment:
Elsewhere we've used lowercase 'council' - not sure which is preferable, but we should probably be consistent.

Reply:
Looks like @DanCorderSoftwire's change (https://github.com/UKGovernmentBEIS/desnz-home-energy-retrofit-beta/pull/45) is removing the need to add 'Council', so I think you can just pass `localAuthorityDetails.Name` :)